src/aux-server/aux-web/shared/scene/decorators/DimensionPositionDecorator.ts

```diff
@@ -383,39 +388,54 @@ export class DimensionPositionDecorator extends AuxBot3DDecoratorBase {
                 }
             }
 
-            // Update bounding objects if trasnformer bot moves.
-            const transformer = getBotTransformer(calc, this.bot3D.bot);
-
-            if (transformer) {
-                const bots =
-                    this.bot3D.dimensionGroup.simulation3D.findBotsById(
-                        transformer
-                    );
-                const parentBot = bots.length ? bots[0] : null;
-
-                if (parentBot instanceof AuxBot3D) {
-                    if (!this._transformerWorldPos) {
-                        this._transformerWorldPos = new Vector3();
-                        parentBot.getWorldPosition(this._transformerWorldPos);
-                        this.bot3D.forceComputeBoundingObjects();
-                    } else {
-                        const lastTransformerPos =
-                            this._transformerWorldPos.clone();
-                        parentBot.getWorldPosition(this._transformerWorldPos);
+            if (update) {
+                this.bot3D.updateMatrixWorld(true);
+            }
+        }
 
-                        if (
-                            !this._transformerWorldPos.equals(
-                                lastTransformerPos
-                            )
-                        ) {
-                            this.bot3D.forceComputeBoundingObjects();
-                        }
-                    }
+        // Update bounding objects if parent moves.
+        // If this bot is parented to a bot or dimension group, crawl up the parent tree until we find it.
+        let parent3D: AuxBot3D | DimensionGroup3D;
+
+        if (
+            this.bot3D.parent.userData.isBotTransformContainer ||
+            this.bot3D.parent.userData.isDimensionGroupDisplay
+        ) {
+            let grandparent = this.bot3D.parent;
+
+            while (!parent3D && grandparent) {
+                if (
+                    grandparent instanceof AuxBot3D ||
+                    grandparent instanceof DimensionGroup3D
+                ) {
+                    parent3D = grandparent;
                 }
+
+                grandparent = grandparent.parent;
             }
+        }
 
-            if (update) {
-                this.bot3D.updateMatrixWorld(true);
+        if (parent3D) {
+            if (!this._parentWorldPos || !this._parentWorldRot) {
+                this._parentWorldPos = new Vector3();
+                this._parentWorldRot = new Quaternion();
+                parent3D.display.getWorldPosition(this._parentWorldPos);
+                parent3D.display.getWorldQuaternion(this._parentWorldRot);
+                this.bot3D.forceComputeBoundingObjects();
+            } else {
+                this._parentWorldPos.copy(tempVector3);
```

Comment:
@Blitzy Does tempVector3 ever get manipulated? From what I'm seeing here, it just stays at its default value forever

Reply:
Yes. The three.js `Vector3.copy` method copies the values of the passed argument. Im doing this just to avoid creating new `Vector3` and `Quaternion` objects every frame when its not necessary.